src/lib/commandline_cmds.ts

```diff
@@ -26,15 +38,25 @@ export function getCommandlineFns(cmdline_state: {
         /**
          * Selects the next completion.
          */
-        next_completion: () => {
+        next_completion: async () => {
+            await awaitProxyEq(
+                contentState,
+                "current_cmdline",
+                "cmdline_filter",
+            )
```

Comment:
How about having a for loop that wraps every function returned by getCommandlineFns in a function that calls `await awaitProxyEq`? This way we don't have to hope we'll remember that some functions need to start with `await awaitProxyEq` when new commandline functions are added.

Reply:
I didn't want to make commands that didn't need to wait for the completions slower. I am particularly concerned about the speed of "insert space or completion".